lib/OptimalBranchingMIS/src/reducer.jl

```diff
@@ -45,30 +45,32 @@ function OptimalBranchingCore.problem_reduce(p::MISProblem, ::MISReducer, TR::Ty
 
         if degmin == 0
             all_zero_vertices = findall(==(0), degrees)
-            return [Branch(MISProblem(remove_vertices(g, all_zero_vertices)), (length(all_zero_vertices)))]
+            return MISProblem(remove_vertices(g, all_zero_vertices)), R(length(all_zero_vertices))
         elseif degmin == 1
-            return [Branch(MISProblem(remove_vertices(g, neighbors(g, vmin) ∪ vmin)), (1))]
+            return MISProblem(remove_vertices(g, neighbors(g, vmin) ∪ vmin)), R(1)
         elseif degmin == 2
             g_new, n = folding(g, vmin)
-            return [Branch(MISProblem(g_new), (n))]
-        elseif degmax ≥ 6
-            return [Branch(MISProblem(remove_vertices(g, closed_neighbors(g, [vmax]))), 1), Branch(MISProblem(remove_vertices(g, [vmax])), 0)]
+            return MISProblem(g_new), R(n)
+        elseif degmax ≥ 6   # Q: No, This is branching!!!!!!!!!
+            error("!!!")
+            return (MISProblem(remove_vertices(g, closed_neighbors(g, [vmax]))), 1), (MISProblem(remove_vertices(g, [vmax])), 0)
         end
     end
 
-    return nothing
+    return p, 0
 end
 
 struct XiaoReducer <: AbstractReducer end
 
-function OptimalBranchingCore.problem_reduce(p::MISProblem, ::XiaoReducer, TR::Type{R}) where R<:AbstractResult
+function OptimalBranchingCore.reduce_problem(::Type{R}, p::MISProblem, ::XiaoReducer, ) where R
     g = p.g
     if nv(g) == 0
-        return [Branch(NoProblem(), 0)]
+        # NOTE: using NoProblem can be slower due to type instability.
+        return MISProblem(SimpleGraph(0)), R(0)
```

Comment:
generate R(0) here is not correct, this gives MISCount(0, 1) and leads to incorrect count.